### PR TITLE
React: Add strictMode option

### PR DESCRIPTION
**Pull request by tooppaaa**
Issue: https://github.com/storybookjs/storybook/issues/12734

## What I did
Add strictMode option

## How to test

- Official Storybook

### CODE REVIEW

File: app/react/src/client/preview/render.tsx [hunk 1]
```diff
@@ -4,15 +4,17 @@ import ReactDOM from 'react-dom';
 
 import { RenderContext } from './types';
 
+// will be provided by the webpack define plugin
+declare const FRAMEWORK_OPTIONS: {
```
Comment (tooppaaa) [inline app/react/src/client/preview/render.tsx:8]: I took a shortcut here duplicating reactOptions type. Typings should be reworked and be under the same rootDir (./src)

**Review by ndelangen — APPROVED** [review]

File: app/react/src/client/preview/render.tsx [hunk 2]
```diff
@@ -47,6 +49,8 @@ class ErrorBoundary extends Component<{
   }
 }
 
+const Wrapper = FRAMEWORK_OPTIONS.strictMode ? React.StrictMode : React.Fragment;
```
Comment (shilman) [inline app/react/src/client/preview/render.tsx:52]:
```suggestion
const isStrict = process.env.STORYBOOK_EXAMPLE_APP || FRAMEWORK_OPTIONS.strictMode;
const Wrapper = isStrict ? React.StrictMode : React.Fragment;
```

**Review by shilman — COMMENTED** [review]
Minor change for backwards compatibility

File: app/react/src/client/preview/render.tsx (hunk 1 above)
Comment (shilman) [reply app/react/src/client/preview/render.tsx:8]: Do you need:

```
import { FRAMEWORK_OPTIONS } from 'global';
```

**Comment by shilman** [comment]
@tooppaaa @tmeasday merging even tho we don't really understand the render count issue

**Pull request merged**